decoder.go

```diff
@@ -155,11 +161,7 @@ func (r *Decoder) decode(typeCode byte) (v interface{}, err error) {
 		if STR_FIXED_START <= typeCode && typeCode < STR_FIXED_START+STR_FIXED_COUNT {
 			b := typeCode - STR_FIXED_START
 			data := make([]byte, b)
-
-			_, err = r.r.Read(data)
-			if err != nil {
-				return
-			}
+			err = r.readIn(data)
 			v = data
```

Comment:
Same as previous comment. Note that this is not a hard Go rule, but rather a good practice used in this (and other) packages